### PR TITLE
Unconfuse web console about resource and kind

### LINKED ISSUE #6990: delete-link, edit-link mix up kind and resource
Applies to all pages, but one example:

Edit link has `kind="services"`, etc. Kind should be upper-case, match the value in the API object (`kind: "Service"`, etc), and be transformed using `kindToResource` before passing to the data service.

Delete link has `resource-type="service"`, which then has an 's' appended to it (`DataService.delete(resourceType + 's', resourceName, context)`). Should probably be using `kind=Service` and `kindToResource` as well.

### LINKED ISSUE #7757: Humanize new repository / image quota types in web console
See #7128

/cc @jwforres

### PR DESCRIPTION
**Pull request by spadgett**
Fixes #6990
Fixes #7757

/cc @jwforres @liggitt

### CODE REVIEW

**Comment by spadgett** [comment]
@jwforres Let me know if you think this is too big for 1.2. I tested every kind. I need the changes to delete-link for deleting HPAs.

File: assets/app/scripts/filters/resources.js [hunk 1]
```diff
@@ -790,28 +790,26 @@ angular.module('openshiftConsole')
       return _.startCase(kind);
     };
   })
-  .filter('humanizeResourceType', function() {
+  .filter('humanizeQuotaResource', function() {
```
Comment (jwforres) [inline assets/app/scripts/filters/resources.js]: assuming nothing else was using humanizeResourceType

Comment (spadgett) [inline assets/app/scripts/filters/resources.js]: right

File: assets/app/scripts/filters/resources.js [hunk 2]
```diff
@@ -790,28 +790,26 @@ angular.module('openshiftConsole')
...
         return resourceType;
       }
 
       var nameFormatMap = {
-        'build': 'Build',
-        'buildconfig': 'Build Config',
-        'deployment': 'Deployment',
-        'deploymentconfig': 'Deployment Config',
-        'imagestream': 'Image Stream',
+        'cpu.limit': 'CPU (Limit)',
```
Comment (derekwaynecarr) [inline assets/app/scripts/filters/resources.js]: limits.cpu

File: assets/app/scripts/filters/resources.js [hunk 3]
```diff
@@ -790,28 +790,26 @@ angular.module('openshiftConsole')
...
+        'cpu.request': 'CPU (Request)',
```
Comment (derekwaynecarr) [inline assets/app/scripts/filters/resources.js]: requests.cpu

File: assets/app/scripts/filters/resources.js [hunk 4]
```diff
@@ -790,28 +790,26 @@ angular.module('openshiftConsole')
...
+        'memory.limit': 'Memory (Limit)',
```
Comment (derekwaynecarr) [inline assets/app/scripts/filters/resources.js]: limits.memory

File: assets/app/scripts/filters/resources.js [hunk 5]
```diff
@@ -790,28 +790,26 @@ angular.module('openshiftConsole')
...
+        'memory.request': 'Memory (Request)',
```
Comment (derekwaynecarr) [inline assets/app/scripts/filters/resources.js]: requests.memory

**Comment by derekwaynecarr** [comment]
We also now have support for ConfigMaps (i.e. configmaps)

**Comment by spadgett** [comment]
thanks @derekwaynecarr, will update

**Comment by jwforres** [comment]
@spadgett does the edit-link not need changes, or was it already using kind?

**Comment by spadgett** [comment]
> does the edit-link not need changes, or was it already using kind?

It's updated in the PR. editLink.js file didn't need any changes, though. editModal.js did.

**Comment by jwforres** [comment]
OK generally LGTM, I think we should go ahead and put this in, just wait till EOD to let the other guys respond about any new quota types

File: assets/app/scripts/filters/resources.js [hunk 6]
```diff
@@ -790,28 +790,26 @@ angular.module('openshiftConsole')
...
+        'openshift.io/projectimagessize': 'Project Image Size',
         'persistentvolumeclaims': 'Persistent Volume Claims',
-        'pod': 'Pod',
         'pods': 'Pods',
-        'project': 'Project',
-        'resourcequotas': 'Resource Quotas',
-        'replicationcontroller': 'Replication Controller',
         'replicationcontrollers': 'Replication Controllers',
+        'resourcequotas': 'Resource Quotas',
         'route': 'Route',
```
Comment (spadgett) [inline assets/app/scripts/filters/resources.js]: @derekwaynecarr Can you quota routes? I didn't see it in the docs. I accidentally left this here.

If so, I assumes it should be `routes`, plural.

**Comment by spadgett** [comment]
@jwforres updated

**Comment by jwforres** [comment]
[merge]

**Pull request merged**